naga-fuzz linking errors on windows msvc

**Description**
I am unable to build wgpu on latest trunk (commit `90859b4e`), using standard `cargo build`. I get linker errors relating to `naga-fuzz`.

**Repro steps**
Clone the repo, run `cargo build`.

**Expected vs observed behavior**
Expected project to build with no errors. Instead got the following linker error:
```
   Compiling naga-fuzz v0.0.0 (C:\Users\benjf\Projects\wgpu\naga\fuzz)
error: linking with `link.exe` failed: exit code: 1561
  |
  = note: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.41.34120\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "C:\\Users\\benjf\\AppData\\Local\\Temp\\rustcntZJDu\\symbols.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.13bkznab1sbnsbbz.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.19ozi22eu2fzogg5.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.1g08iljx79a0bt9e.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.21g6l22s43cr2hw1.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.2esfdbdfn6xg105h.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.2py6suelwhnqk5j3.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.2zx7snnbrn021nic.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.37hwgqnap4jbl7fv.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.3b4sbpgevioaaizn.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.3bs8p0vo29id600.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.3eqwhhx18c5421yw.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.3grs4cltbv6u1i2e.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.3hhvpf39pnllj7xc.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.3op380gsnyjrhazs.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.40ajhjag5moyqpnc.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.4bxhzb7anrbsdmvv.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.4debtunp6wp9n5qb.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.4j6yni6o6dgzndd5.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.4vvax1vwpzzpktq9.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.54037w5dd4kw8vo7.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.55bt2dxd8duh7rm2.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.56zx8fjhvpftzu9k.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.59ifw94erszw6v0p.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.5blccagc5ulmi82w.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.5cuficn8nu27f104.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.5gjkqlucf9qlyrvo.rcgu.o" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.8z5u35fwwfl87nt.rcgu.o" "/LIBPATH:C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps" "/LIBPATH:C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\build\\libfuzzer-sys-b42cd1fa791ba6e4\\out" "/LIBPATH:C:\\Users\\benjf\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\windows_x86_64_msvc-0.48.5\\lib" "/LIBPATH:C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libnaga-9060678b922abed2.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\librustc_hash-10e67c6ddcf1011a.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libhexf_parse-9c8b0686ca88af2a.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\liblog-544a9432648ec6b4.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libbit_set-6ff8ef9ddda2d280.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libbit_vec-71ce6c75921a1eb5.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libarrayvec-88d891c010ce6c70.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libpetgraph-67aa1ac7dd8405dc.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libfixedbitset-5012b99a4b7f6132.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libpp_rs-dcf7e427837c0098.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libunicode_xid-9bf43483bc4eae27.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libcodespan_reporting-c82982eb947cb474.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libunicode_width-d7cf3ff72900502a.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libtermcolor-db328313a6ccc268.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libwinapi_util-fc14ebbf1b7a2749.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libwindows_sys-92b9ece4a9886c5c.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libwindows_targets-5ed392ba99a296da.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libindexmap-6ff8b668ca7e9b03.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libequivalent-6c1f94c3084b293f.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libhashbrown-bae5fc46570dab84.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libspirv-6c611c9664bff619.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libbitflags-c3ab2eeb345fdfc6.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libserde-5cb19b7b630761d0.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libthiserror-44d5b331fd77c297.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\liblibfuzzer_sys-d6f612dee6a18e7c.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libonce_cell-08067d874d72c3dc.rlib" "C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\libarbitrary-333c54ea27a6d204.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-e493bcbfdc66a475.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-467acea86c440d1f.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_demangle-8bcec20f8d868561.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd_detect-e75d865d889e433e.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libhashbrown-cd24a5810f58b720.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-44b4ecbbc95939b2.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-ff93d2b34eb6aecc.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcfg_if-17346c417f0e9166.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-ff5a06ebf4571d10.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-cb2478631e21007b.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_core-58d59322b34f2b51.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-c192803e6895f627.rlib" "C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-20106e86b5dab94f.rlib" "windows.0.48.5.lib" "kernel32.lib" "advapi32.lib" "bcrypt.lib" "kernel32.lib" "ntdll.lib" "userenv.lib" "ws2_32.lib" "kernel32.lib" "ws2_32.lib" "kernel32.lib" "ntdll.lib" "kernel32.lib" "msvcrt.lib" "/NXCOMPAT" "/LIBPATH:C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/OUT:C:\\Users\\benjf\\Projects\\wgpu\\target\\debug\\deps\\ir.exe" "/OPT:REF,NOICF" "/DEBUG" "/NATVIS:C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:C:\\Users\\benjf\\.rustup\\toolchains\\1.76-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis"
  = note:    Creating library C:\Users\benjf\Projects\wgpu\target\debug\deps\ir.lib and object C:\Users\benjf\Projects\wgpu\target\debug\deps\ir.exp
          LINK : fatal error LNK1561: entry point must be defined
```

**Platform**
I'm running windows 11, with Microsoft Visual Studio Community 2022.

--- a/naga/fuzz/build.rs
+++ b/naga/fuzz/build.rs
@@ -1,5 +1,6 @@
 fn main() {
     cfg_aliases::cfg_aliases! {
-        enable_fuzzing: { not(any(target_arch = "wasm32", target_os = "ios", all(windows, target_arch = "aarch64"))) },
+        fuzzable_platform: { not(any(target_arch = "wasm32", target_os = "ios", all(windows, target_arch = "aarch64"))) },
     }
+    println!("cargo::rustc-check-cfg=cfg(fuzzing)");
 }

--- a/naga/fuzz/build.rs
+++ b/naga/fuzz/build.rs
@@ -2,5 +2,6 @@ fn main() {
     cfg_aliases::cfg_aliases! {
         fuzzable_platform: { not(any(target_arch = "wasm32", target_os = "ios", all(windows, target_arch = "aarch64"))) },
     }
+    // This cfg provided by cargo-fuzz
     println!("cargo::rustc-check-cfg=cfg(fuzzing)");
 }

--- a/naga/fuzz/fuzz_targets/glsl_parser.rs
+++ b/naga/fuzz/fuzz_targets/glsl_parser.rs
@@ -1,6 +1,6 @@
-#![cfg_attr(enable_fuzzing, no_main)]
+#![cfg_attr(all(fuzzable_platform, fuzzing), no_main)]
 
-#[cfg(enable_fuzzing)]
+#[cfg(all(fuzzable_platform, fuzzing))]
 mod fuzz {
     use std::iter::FromIterator;
 
@@ -40,5 +40,5 @@ mod fuzz {
     });
 }
 
-#[cfg(not(enable_fuzzing))]
+#[cfg(not(all(fuzzable_platform, fuzzing)))]
 fn main() {}

--- a/naga/fuzz/fuzz_targets/ir.rs
+++ b/naga/fuzz/fuzz_targets/ir.rs
@@ -1,6 +1,6 @@
-#![cfg_attr(enable_fuzzing, no_main)]
+#![cfg_attr(all(fuzzable_platform, fuzzing), no_main)]
 
-#[cfg(enable_fuzzing)]
+#[cfg(all(fuzzable_platform, fuzzing))]
 mod fuzz {
     use libfuzzer_sys::fuzz_target;
 
@@ -14,5 +14,5 @@ mod fuzz {
     });
 }
 
-#[cfg(not(enable_fuzzing))]
+#[cfg(not(all(fuzzable_platform, fuzzing)))]
 fn main() {}

--- a/naga/fuzz/fuzz_targets/spv_parser.rs
+++ b/naga/fuzz/fuzz_targets/spv_parser.rs
@@ -1,6 +1,6 @@
-#![cfg_attr(enable_fuzzing, no_main)]
+#![cfg_attr(all(fuzzable_platform, fuzzing), no_main)]
 
-#[cfg(enable_fuzzing)]
+#[cfg(all(fuzzable_platform, fuzzing))]
 mod fuzz {
     use libfuzzer_sys::fuzz_target;
     use naga::front::spv::{Frontend, Options};
@@ -12,5 +12,5 @@ mod fuzz {
     });
 }
 
-#[cfg(not(enable_fuzzing))]
+#[cfg(not(all(fuzzable_platform, fuzzing)))]
 fn main() {}

--- a/naga/fuzz/fuzz_targets/wgsl_parser.rs
+++ b/naga/fuzz/fuzz_targets/wgsl_parser.rs
@@ -1,6 +1,6 @@
-#![cfg_attr(enable_fuzzing, no_main)]
+#![cfg_attr(all(fuzzable_platform, fuzzing), no_main)]
 
-#[cfg(enable_fuzzing)]
+#[cfg(all(fuzzable_platform, fuzzing))]
 mod fuzz {
     use libfuzzer_sys::fuzz_target;
     use naga::front::wgsl::Frontend;
@@ -11,5 +11,5 @@ mod fuzz {
     });
 }
 
-#[cfg(not(enable_fuzzing))]
+#[cfg(not(all(fuzzable_platform, fuzzing)))]
 fn main() {}

